src/main/java/org/jabref/gui/collab/ExternalChangesResolverViewModel.java

```diff
@@ -100,8 +102,15 @@ public void acceptMergedChange(DatabaseChange databaseChange) {
     }
 
     public void applyChanges() {
-        changes.stream().filter(DatabaseChange::isAccepted).forEach(change -> change.applyChange(ce));
-        ce.end();
-        undoManager.addEdit(ce);
+        if (changes.stream().noneMatch(change -> !change.isAccepted())) {
```

Comment:
Isn't it that all changes are accepted? Then, it can be formulated more directly:

```suggestion
        if (changes.stream().allMatch(change -> change.isAccepted())) {
```